control-plane-operator/controllers/hostedcontrolplane/olm/catalogs.go

```diff
@@ -153,6 +159,24 @@ func ReconcileCatalogsImageStream(imageStream *imagev1.ImageStream, ownerRef con
 	return nil
 }
 
+// getCatalogToImageWithOpenShiftImageRegistryOverrides returns a map of
+// images top be used for the catalog registries where the image address got
+// amended according to OpenShiftImageRegistryOverrides as set on the HostedControlPlaneReconciler
+func getCatalogToImageWithOpenShiftImageRegistryOverrides(catalogToImage map[string]string, openShiftImageRegistryOverrides map[string][]string) map[string]string {
+	catalogWithOverride := make(map[string]string)
+	for name, image := range catalogToImage {
+		for registrySource, registryDest := range openShiftImageRegistryOverrides {
+			if strings.Contains(image, registrySource) {
```

Comment:
instead of doing a string check, I'd rather parse the image ref and match the registry exactly, using reference.Parse() from
https://github.com/openshift/hypershift/blob/8b4b52925d47373f3fe4f0d5684c88dc8a93368a/support/thirdparty/library-go/pkg/image/reference/reference.go#L32
then replace the registry in the reference object

Reply:
This is a bit ambiguous: `reference.Parse("registry.redhat.io/redhat/redhat-operator-index:v4.14")` will return a DockerImageReference where `Registry=registry.redhat.io` and `Namespace=redhat` but (probably) we don't want to replace just the registry forcing the customers to keep using the same namespace in their mirrored registry.

And indeed in the cmd help of the control-plane-operator (see: https://github.com/openshift/hypershift/blob/main/control-plane-operator/main.go#L177 ) we say:
`Images before being applied are scanned for the source registry string and if found the string is replaced with the destination registry string.`

which is exactly what is done in: https://github.com/openshift/hypershift/blob/068f6427aab4632556d250377d51b0e3422998cf/support/releaseinfo/registry_image_content_policies.go#L26-L28
I'm not directly calling that method since here we lack the `pullSecret` that is used to validate the replacement upfront, but at least the implementation is coherent.